Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/SparkJobHttpHandler.java

```diff
@@ -0,0 +1,82 @@
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+public class SparkJobHttpHandler implements HttpHandler {
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+        httpExchange.getResponseHeaders().add("Access-Control-Allow-Origin", "*");
+        JobRequestDetails requestDetail = JobRequestDetails.getJobRequestDetail(httpExchange);
+        // TODO : get message from spark history server
+        try {
+            String path = requestDetail.getRequestPath();
+            if (path.equalsIgnoreCase("/applications/") && requestDetail.getAppId().equalsIgnoreCase("0")) {
+                List<Application> applications = JobViewCacheManager.getSparkApplications(requestDetail.getCluster());
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            } else if (path.contains("application_graph")) {
+                ApplicationKey key = new ApplicationKey(requestDetail.getCluster(), requestDetail.getAppId());
+                List<Job> jobs = JobViewCacheManager.getJob(key);
+                App app = JobViewCacheManager.getYarnApp(key);
+                YarnAppWithJobs yarnAppWithJobs = new YarnAppWithJobs(app, jobs);
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(yarnAppWithJobs);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            } else if (path.contains("stages_summary")) {
+                List<Stage> stages = JobViewCacheManager.getStages(new ApplicationKey(requestDetail.getCluster(), requestDetail.getAppId()));
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(stages);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            } else if (path.contains("executors_summary")) {
+                List<Executor> executors = JobViewCacheManager.getExecutors(new ApplicationKey(requestDetail.getCluster(), requestDetail.getAppId()));
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(executors);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            } else if (path.contains("tasks_summary")) {
+                List<Task> tasks = JobViewCacheManager.getTasks(new ApplicationKey(requestDetail.getCluster(), requestDetail.getAppId()));
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(tasks);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            }
```

Comment:
An 404 error can be expected if nothing matched. I think it's enough for debugging. It's an internal http server after all.